ui/src/utils/index.ts

```diff
@@ -11,106 +11,106 @@ const { message } = createDiscreteApi(['message']);
+  let minuteLabel = t('Minute');
 
-    if (item > 1) {
-        minuteLabel = t('Minutes');
-    }
+  if (item > 1) {
+    minuteLabel = t('Minutes');
+  }
 
-    return `${item} ${minuteLabel}`;
+  return `${item} ${minuteLabel}`;
 };
 
 export const writeBufferToTerminal = (
-    enableZmodem: boolean,
-    zmodemStatus: boolean,
-    terminal: Terminal | null,
-    data: any
+  enableZmodem: boolean,
+  zmodemStatus: boolean,
+  terminal: Terminal | null,
+  data: any
 ) => {
-    if (!enableZmodem && zmodemStatus) return message.error('未开启 Zmodem 且当前在 Zmodem 状态, 不允许显示');
+  if (!enableZmodem && zmodemStatus) return message.error('未开启 Zmodem 且当前在 Zmodem 状态, 不允许显示');
 
-    terminal && terminal.write(new Uint8Array(data));
+  terminal && terminal.write(new Uint8Array(data));
 };
 
 export const preprocessInput = (data: string, config: ILunaConfig) => {
-    // 如果配置项 backspaceAsCtrlH 启用（值为 "1"），并且输入数据包含删除键的 ASCII 码 (AsciiDel，即 127)，
-    // 它会将其替换为退格键的 ASCII 码 (AsciiBackspace，即 8)
-    if (config.backspaceAsCtrlH === '1') {
-        if (data.charCodeAt(0) === AsciiDel) {
-            data = String.fromCharCode(AsciiBackspace);
-        }
+  // 如果配置项 backspaceAsCtrlH 启用（值为 "1"），并且输入数据包含删除键的 ASCII 码 (AsciiDel，即 127)，
+  // 它会将其替换为退格键的 ASCII 码 (AsciiBackspace，即 8)
+  if (config.backspaceAsCtrlH === '1') {
+    if (data.charCodeAt(0) === AsciiDel) {
+      data = String.fromCharCode(AsciiBackspace);
     }
+  }
 
-    // 如果配置项 ctrlCAsCtrlZ 启用（值为 "1"），并且输入数据包含 Ctrl+C 的 ASCII 码 (AsciiCtrlC，即 3)，
-    // 它会将其替换为 Ctrl+Z 的 ASCII 码 (AsciiCtrlZ，即 26)。
-    if (config.ctrlCAsCtrlZ === '1') {
-        if (data.charCodeAt(0) === AsciiCtrlC) {
-            data = String.fromCharCode(AsciiCtrlZ);
-        }
-    }
-
-    if (data === '^C') {
-        data = '\x03';
+  // 如果配置项 ctrlCAsCtrlZ 启用（值为 "1"），并且输入数据包含 Ctrl+C 的 ASCII 码 (AsciiCtrlC，即 3)，
+  // 它会将其替换为 Ctrl+Z 的 ASCII 码 (AsciiCtrlZ，即 26)。
+  if (config.ctrlCAsCtrlZ === '1') {
+    if (data.charCodeAt(0) === AsciiCtrlC) {
+      data = String.fromCharCode(AsciiCtrlZ);
     }
+  }
 
+  if (data.includes('\u001b[200~') || data.includes('\u001b[201~')) {
+    return data.replace(/\u001b\[200~|\u001b\[201~/g, '');
+  } else {
     return data;
+  }
 };
 
 export const getOs = () => {
-    const { platform, userAgent } = navigator;
+  const { platform, userAgent } = navigator;
 
-    if (platform.startsWith('Win') || userAgent.includes('Windows')) {
-        return 'Windows';
-    }
+  if (platform.startsWith('Win') || userAgent.includes('Windows')) {
+    return 'Windows';
+  }
 
-    if (platform.startsWith('Mac') || userAgent.includes('Macintosh')) {
-        return 'macOS';
-    }
+  if (platform.startsWith('Mac') || userAgent.includes('Macintosh')) {
+    return 'macOS';
+  }
 
-    if (platform.startsWith('Linux') || userAgent.includes('Linux')) {
-        return 'Linux';
-    }
+  if (platform.startsWith('Linux') || userAgent.includes('Linux')) {
+    return 'Linux';
+  }
 };
```

Comment:
The code does not contain any issues but it might be slightly optimized:

- Instead of checking and catching the error when writing directly to `terminal` using an array, use `Set`.

Here is the recommended way of doing this:
```python
const terminal = new Terminal();
terminal.on('error', () => console.error('Error'));
setImmediate(function() {
    for (let i = 0; i < text.length; ++i) {
        const chrCode = String.fromCodePoint(text.charCodeAt(i));
        terminal.write(chrCode);
    }

    process.exit(0); // exit immediately.
});
```
The function should call `process.exit(0)` at the end to ensure that the main thread stops running. The immediate callback runs after every character, so you do not have to worry about asynchronous operations.

Note how I've introduced `console.error()` instead of direct logging calls to the application's own logger framework, just like JavaScript recommends. In production applications, consider adding some level of error handling within the app logic itself.
```javascript
try {
  if (disableLogging) throw new Error('Disabled Logging');
  
  const terminal = new Terminal();
  process.on('unhandledRejection', err => {
    console.error(err.stack || err);
  });
  
  setImmediate(() => {
    const lines = [];
    text.split('').forEach(line => {
      line.chars.forEach(charcode =>
         terminal.write(String.fromCharCode(parseInt(charcode)))
       );

      lines.push('');
    });

    process.exit(lines.join('\n')); // exit immediately.
  }, 0);
} catch ({ stack }) {

}
```

I also included a placeholder for logging errors with unhandled rejections. Always handle uncaught exceptions or undefined behavior more carefully in complex systems rather than relying on logs, though these are useful tools in debugging scripts. For real-world usage, include proper error handling logic inside your handlers.